src/main/java/org/scijava/jupyter/notebook/DefaultNotebookService.java

```diff
@@ -56,9 +56,30 @@
     @Parameter
     private LogService log;
 
-    public DefaultNotebookService() {
-        Displayers.register(String.class, (Displayer) StringDisplayer.get());
-        Displayers.register(List.class, (Displayer) ListDisplayer.get());
+    @Parameter
+    private ConvertService converService;
+
+    /**
+     * Use the most appropriate output type according to the input object.
+     *
+     * @param object
+     * @return
+     */
+    @Override
+    public Object display(Object object) {
+
+        if (object.getClass() == String.class) {
```

Comment:
Instead of hardcoding this: why not simply type the converters to produce `NotebookOutput` generically. Then you can write `convertService.convert(object, NotebookOutput.class)` agnostically without the case logic here. As long as each kind of object has a "preferred" (i.e., highest priority) type of `NotebookOutput` that it will actually produce, this should work well. The only problem will be the case you raise where we want the user to select something; that is a separate issue though which is also not solved by the hardcoded case logic here.